bakery/baker-client/src/main/scala/com/ing/bakery/common/FailoverUtils.scala

```diff
@@ -26,25 +25,25 @@ object FailoverUtils extends LazyLogging{
     * @return
     */
   def calWithFailOver(fos: FailoverState,
-                                   client: Client[IO],
-                                   func: Uri => IO[Request[IO]],
-                                   filters: Seq[Request[IO] => Request[IO]],
-                                   handleHttpErrors: Response[IO] => IO[Throwable])
-                                  (implicit ec: ExecutionContext, decoder: Decoder[BakerResult]): IO[BakerResult] = {
-
+                      client: Client[IO],
+                      func: Uri => IO[Request[IO]],
+                      filters: Seq[Request[IO] => Request[IO]],
+                      handleHttpErrors: Response[IO] => IO[Throwable])
+                     (implicit ec: ExecutionContext, decoder: Decoder[BakerResult]): IO[BakerResult] = {
     implicit val timer: Timer[IO] = IO.timer(ec)
 
-    val call = client
-      .expectOr(func(fos.host).map({ request: Request[IO] =>
-        filters.foldLeft(request)((acc, filter) => filter(acc))
-      }))(handleHttpErrors)(jsonOf[IO, BakerResult])
+    def call(uri: Uri): IO[BakerResult] =
+      client
+        .expectOr(func(uri).map({ request: Request[IO] =>
+          filters.foldLeft(request)((acc, filter) => filter(acc))
+        }))(handleHttpErrors)(jsonOf[IO, BakerResult])
 
     retryingOnAllErrors(
       policy = limitRetries[IO](fos.size * 2) |+| exponentialBackoff[IO](5.milliseconds),
-      onError = (error: Throwable, _: RetryDetails) => IO {
-        logger.warn(s"Failed to call host ${fos.host}", error)
+      onError = (error: Throwable, retryDetails: RetryDetails) => IO {
+        logger.warn(s"Failed to call host ${fos.host}, retry #${retryDetails.retriesSoFar}", error)
```

Comment:
will the error be logged here with full stacktrace? maybe we don't need it - just `error.getMessage` ?

Reply:
Agree, decreased the logs pollution.